python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -285,15 +288,16 @@ def test_list_with_non_list(seq):
 @pytest.mark.parametrize("factory", [
     pa.list_, pa.large_list, pa.list_view, pa.large_list_view])
 def test_nested_arrays(seq, factory):
-    arr = pa.array(seq([np.array([], dtype=np.int64),
-                        np.array([1, 2], dtype=np.int64), None]),
+    arr = pa.array(seq([pa.array([], type=pa.int64()),
+                        pa.array([1, 2], type=pa.int64()), None]),
```

Comment:
In hindsight, this should maybe be kept as numpy arrays .. (and just mark as requiring numpy)  
At least it is useful to also test nested data construction where the inner values are numpy arrays (probably ideally we would test both, like parametrize on the sequence type for the inner sequences as well?)

Reply:
I've parametrized the test with inner_sequence reusing the `sequence_types` defined